src/taskpane/pages/dashboard.tsx

```diff
@@ -103,11 +124,21 @@ function Dashboard({ citeSupport }: DashboardProps): ReactElement {
     }
   }
 
-  const checkItems = (itemIds: Array<string>) => {
+  const checkItems = (itemsMetadata: Array<CitationItem>) => {
```

Comment:
This method is now no longer about marking items selected, so it should be renamed. But actually, it seems like a good idea to split the method in a "updateMetada" (which you already added further below) and "checkItems" method, each having a clearly defined single purpose.

Reply:
If we split this method, we have to update items twice, which means we are re-rendering the component twice, and rendering is kinda expensive.